src/Background/stat_estimators.jl

```diff
@@ -1,5 +1,44 @@
 using Statistics
 
+"""
+    sigma_clip(data, sigma_low, sigma_high; center=median, std=std, iterations=5)
+    sigma_clip(data, sigma; center=median, std=std, iteration=5)
+
+This function returns sigma clipped values of the input `data`. The funtion `sigma_clip!` is an inplace implementation and mutates the `data`.
+`sigma_high` and `sigma_low` are for un-symmetrical clipping, when `sigma_low = sigma_high` then they can be passed as `sigma`.
+`center`, `std` and `iterations` are optional parameters which are functions for finding central element, standard deviation and number of iterations of sigma
+clipping to be carried out.
+
+# Example
+```jldoctest
+julia> data = [1, 2, 3];
+
+julia> sigma_clip(data, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+
+julia> sigma_clip(data, 1, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+```
+"""
+
+function sigma_clip!(data::AbstractArray, sigma_low::Real, sigma_high::Real; iterations::Integer=5, center=median, std=std)
+    for i=1:iterations
+        mean = center(data)
+        deviation = std(data)
+        clamp!(data, mean - sigma_low * deviation, mean + sigma_high * deviation)
+    end
+    return data
+end
+
+sigma_clip!(data, sigma; kwargs...) = sigma_clip!(data, sigma, sigma; kwargs...)
+sigma_clip(data, rest...; kwargs...) = sigma_clip!(copy(data), rest...; kwargs...)
```

Comment:
I believe that if you do
```suggestion
sigma_clip(data, rest...; kwargs...) = sigma_clip!(float(copy(data)), rest...; kwargs...)
```
this function can take as input also integers, so that in the tests you can use
```julia
y = [1, 2, 3, 4, 5, 6] 
```

Reply:
my standard usage is
```julia
function my_favorite_subroutine!(A)
    # implementation
    return output
end
my_favorite_subroutine(A) = my_favorite_subroutine!(deepcopy(A))
```
As for the type stability, if we're passing the array into float we can probably just do
```sigma_clip!(float(data), rest...;)``` if that makes sense